source/extensions/resource_monitors/cpu_utilization/cpu_utilization_monitor.cc

```diff
@@ -40,20 +44,24 @@ void CpuUtilizationMonitor::updateResourceUsage(Server::ResourceUpdateCallbacks&
     callbacks.onFailure(error);
     return;
   }
-  const double work_over_period = cpu_times.work_time - previous_cpu_times_.work_time;
-  const int64_t total_over_period = cpu_times.total_time - previous_cpu_times_.total_time;
-  if (work_over_period < 0 || total_over_period <= 0) {
-    const auto& error = EnvoyException(
-        fmt::format("Erroneous CPU stats calculation. Work_over_period='{}' cannot "
-                    "be a negative number and total_over_period='{}' must be a positive number.",
-                    work_over_period, total_over_period));
-    callbacks.onFailure(error);
+
+  // Calculate utilization (encapsulates cgroup v1 vs v2 differences and validation)
+  double current_utilization;
+  TRY_ASSERT_MAIN_THREAD {
```

Comment:
I think we can get rid of the exception here if the `calculateUtilization` returned an `absl::Status` as well